### PR TITLE
Patch security

### PR DESCRIPTION
**Pull request by gparant**
Use access token to get auth token connection

### CODE REVIEW

File: play/src/front/Connexion/LocalUserStore.ts [hunk 1]
```diff
@@ -105,8 +106,13 @@ class LocalUserStore {
...
 
     getCharacterLayers(): string[] | null {
-        const value = JSON.parse(localStorage.getItem(characterLayersKey) || "null");
-        return areCharacterLayersValid(value) ? value : null;
+        try {
+            const value = JSON.parse(localStorage.getItem(characterLayersKey) || "null");
+            return areCharacterLayersValid(value) ? value : null;
+        } catch (err) {
+            console.error("LocalUserStore error: ", err);
+            return null;
```
Comment (moufmouf) [inline play/src/front/Connexion/LocalUserStore.ts:114]: This does not seem related to the PR. What are we trying to solve here?

File: play/src/front/Connexion/ConnectionManager.ts [hunk 2]
```diff
@@ -88,12 +89,44 @@ class ConnectionManager {
...
+            if (ADMIN_URL != undefined && accessToken != undefined) {
+                try {
+                    const res = await Axios.get(`${ADMIN_URL}/api/workadventure/privatetoken/${accessToken}`);
+                    console.log("res", res);
+                    accessToken = res.data.accessToken;
+                    token = res.data.privateToken;
+                } catch (e) {
+                    console.error("Init game connection get private token", e);
+                }
+            }
```
Comment (moufmouf) [inline play/src/front/Connexion/ConnectionManager.ts:115]: The TODO is correct here.

You need to make this call from the pusher.

**Review by moufmouf — COMMENTED** [review]
You should probably make the call to `[admin]/api/workadventure/privatetoken` from the pusher.
This route should probably be protected by the ADMIN_API_TOKEN too.

I think you can do everything from the `FrontController` in the pusher instead of doing this in the front. Here, you could analyze the URL, check if there is an "accessToken" parameter in the URL. If there is one, do the request on the admin, fetch the "accessToken" and pass it back to the front via a JS variable embedded in the HTML.

Also, maybe we should rename the "?accessToken=" GET parameter, because it is not a real OAuth accessToken. It is just a temporary identifier that allows us to fetch the OAuth accessToken.

So maybe we should rename this to something that is less confusing? (I don't know exactly what... `tempToken`? `exchangeToken`?

Finally, do not forget to do the request to the Admin from the `AdminApi.ts` file and to update all the (Zod) types to have the new endpoint appear clearly in the OpenAPI documentation.

**Comment by gparant** [comment]
Update strategy in #2929
